mypy/stubtest.py

```diff
@@ -243,6 +259,60 @@ def _belongs_to_runtime(r: types.ModuleType, attr: str) -> bool:
         )
 
 
+IGNORED_DUNDERS = frozenset({
+    # Very special attributes
+    "__weakref__",
+    "__slots__",
+    "__dict__",
+    "__text_signature__",
+    # Pickle methods
+    "__setstate__",
+    "__getstate__",
+    "__getnewargs__",
+    "__getinitargs__",
+    "__reduce_ex__",
+    "__reduce__",
+    # typing implementation details
+    "__parameters__",
```

Comment:
I think I'd prefer the typing ones to be in typeshed allowlists, but don't feel strongly if you disagree (we can regex match in the allowlists with `typing\..*\.__(parameters|origin|...)__)`

Reply:
I think I'll take `__forward_is_class__`, `__forward_module__` and `__mro_entries__` out -- the first two are very specific to `ForwardRef`, and the third could conceivably have some usages outside of `typing`. But I'd rather leave the others in, since they are _purely_ typing implementation details and have the potential to show up in a _lot_ of classes if somebody in third-party code is inheriting from `Generic`.

Reply:
Sounds good, I don't feel strongly here. Although maybe these should actually be defined on Generic? (The Python version history of these is a little complicated, but) e.g. `__origin__` and `__args__` are very popular for runtime typing use cases, stubtest included!

Reply:
Actually, let's just revisit this later, like `__match_args__`

Reply:
Maybe you should be put on the typing naughty step if you're accessing `__origin__` and `__args__` directly instead of using `get_origin` and `get_args`? ;)